django/db/models/base.py

```diff
@@ -1708,7 +1708,10 @@ def _check_ordering(cls):
             fld = None
             for part in field.split(LOOKUP_SEP):
                 try:
-                    fld = _cls._meta.get_field(part)
+                    if part == 'pk' and not fld == _cls._meta.pk:
+                        fld = _cls._meta.pk
+                    else:
+                        fld = _cls._meta.get_field(part)
```

Comment:
It seems that we have two issues here, i.e. you can use fields from the same model multiple times, e.g. `parent__field1__field2__pk__field1`, and you cannot use `pk`. I think we should clean `_cls` if a field is not relation, e.g.
```python
if part == 'pk':
    fld = _cls._meta.pk
else:
    fld = _cls._meta.get_field(part)
if fld.is_relation:
    _cls = fld.get_path_info()[-1].to_opts.model
else:
    _cls = None
```
I would split this into two fixes, first for using multiple times fields from the same model (with test):
```
fld = _cls._meta.get_field(part)
if fld.is_relation:
    _cls = fld.get_path_info()[-1].to_opts.model
else:
    _cls = None
```
and second to handle `pk` (with test).

Reply:
Thanks for the comment.
I separated the changes in two PR as you said.